chgnet/model/dynamics.py

```diff
@@ -89,7 +89,7 @@ def __init__(
                 self.device = f"cuda:{cuda_devices_sorted_by_free_mem()[-1]}"
 
         # Move the model to the specified device
-        self.model = (model or CHGNet.load()).to(self.device)
+        self.model = (model or CHGNet.load().float()).to(self.device)
```

Comment:
Not sure it's generally a good idea to lower model precision? But I can't use the model on MPS without it.

Should we restrict the precision lowering to MPS?

Reply:
Never mind, not actually an issue with CHGNet. Was caused by running MACE and CHGNet in the same session. [MACE calls `torch.set_default_dtype(torch.float64)`](https://github.com/acesuit/mace/blob/6c6e9a447621b6a3f4a3f49333c1b0ba2e94bf4f/mace/calculators/mace.py#L151), affecting all other torch models used concurrently.

@ilyes319 Can we make `dtype` fully local to MACE to make it compatible with other models?